tests/scripts/rm_hpo_api_tests.sh

```diff
@@ -328,8 +329,258 @@ function other_post_experiment_tests() {
+			tunable_name=$(cat ${parse_json} | jq '.['${tunable_count}'].name')
+			actual_tunable_name=$(cat ${result} | jq '.['${tunable_count}'].tunable_name')
+			actual_tunable_value=$(cat ${result} | jq '.['${tunable_count}'].tunable_value')
+
+			echo "" | tee -a ${LOG_} ${LOG}
+			echo "Validating the tunable name ${actual_tunable_name}..." | tee -a ${LOG_} ${LOG}
+			# validate the tunable name
+			validate_tunable_name
+			echo "" | tee -a ${LOG_} ${LOG}
+
+			echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~" | tee -a ${LOG_} ${LOG}
+			echo "" | tee -a ${LOG_} ${LOG}
+			echo "Validating the tunable value for ${actual_tunable_name}..." | tee -a ${LOG_} ${LOG}
+			# validate the tunable value
+			validate_tunable_value
+			echo "" | tee -a ${LOG_} ${LOG}
+			echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~" | tee -a ${LOG_} ${LOG}
+			((tunable_count++))
+		done
+	fi
+	echo ""
+}
+
+# Validate the trial JSON returned by RM-HPO GET API
+# input: test name
+function get_trial_json_valid_tests() {
+	__test_name__=$1
+
+	IFS=' ' read -r -a get_trial_json_valid_tests <<<  ${rm_hpo_get_trial_json_tests[$FUNCNAME]}
+	for exp_trial in "${get_trial_json_valid_tests[@]}"
+	do
+		TESTS_="${TEST_DIR}/${FUNCNAME}"
+		mkdir -p ${TESTS_}
+		LOG_="${TEST_DIR}/${FUNCNAME}.log"
+		result="${TESTS_}/${exp_trial}_result.log"
+		parse_json="${TESTS_}/${exp_trial}_actual_json.json"
+		echo "************************************* ${exp_trial} Test ****************************************" | tee -a ${LOG_} ${LOG}
+	
+		# Start the HPO servers
+		${SCRIPTS_DIR}/start_hpo_servers.sh -p ${TESTS_} | tee -a ${LOG_} ${LOG}
+		
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+	
+		# Check if the servers have started
+		check_server_status
+			
+		# Get the id from search space JSON
+		current_id=$(cat ${SEARCH_SPACE_JSON} | jq .[].id | tr -d '""')
+		
+		if [ "${exp_trial}" == "valid-exp-trial" ]; then
+			create_post_exp_json_array ${current_id}
+			post_experiment_json "${rm_hpo_post_experiment_json[$experiment]}"
+			trial_num="${response}"
+		else
+			exp="valid-experiment"	
+			operation_generate_subsequent
+			trial_num="${response}"
+		fi
+		
+		run_get_trial_json_test "valid-exp-trial" "${trial_num}"
+
+		actual_result="${http_code}"
+		
+		expected_result_="200"
+		expected_behaviour="RESPONSE_CODE = 200 OK"
+
+		compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+		
+		if [ "${failed}" -eq 0 ]; then
+			validate_exp_trial
+		fi
+		
+		# Stop the HPO servers
+		${SCRIPTS_DIR}/start_hpo_servers.sh -t | tee -a ${LOG_} ${LOG}
+		echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
+	done
+}
+
 # Tests for RM-HPO POST experiment
 function rm_hpo_post_experiment() {
-	invalid_post ${FUNCNAME}
+	invalid_post_experiment_tests ${FUNCNAME}
```

Comment:
This seems to have a valid experiment too, so rename invalid to run. Also change the name of the array